sql/src/main/java/io/druid/sql/calcite/schema/DruidSchema.java

```diff
@@ -296,26 +295,39 @@ private DruidTable computeTable(final String dataSource)
       return null;
     }
 
-    final Map<String, ColumnAnalysis> columnMetadata = Iterables.getOnlyElement(results).getColumns();
-    final RowSignature.Builder rowSignature = RowSignature.builder();
+    final Map<String, ValueType> columnTypes = Maps.newLinkedHashMap();
 
-    for (Map.Entry<String, ColumnAnalysis> entry : columnMetadata.entrySet()) {
-      if (entry.getValue().isError()) {
-        // Ignore columns with metadata consistency errors.
-        continue;
-      }
+    // Resolve conflicts by taking the latest metadata. This aids in gradual schema evolution.
+    String maxSegmentId = "";
 
-      ValueType valueType;
-      try {
-        valueType = ValueType.valueOf(entry.getValue().getType().toUpperCase());
-      }
-      catch (IllegalArgumentException e) {
-        // Assume unrecognized types are some flavor of COMPLEX. This throws away information about exactly
-        // what kind of complex column it is, which we may want to preserve some day.
-        valueType = ValueType.COMPLEX;
+    for (SegmentAnalysis analysis : results) {
+      for (Map.Entry<String, ColumnAnalysis> entry : analysis.getColumns().entrySet()) {
+        if (entry.getValue().isError()) {
+          // Skip columns with analysis errors.
+          continue;
+        }
+
+        if (!columnTypes.containsKey(entry.getKey()) || analysis.getId().compareTo(maxSegmentId) > 0) {
```

Comment:
Hmm, I was thinking:

- I want to avoid specifying any analysis types like INTERVAL, for performance reasons
- Segment ID should be close enough for this purpose, it should be equivalent to comparing start instant

But it looks like INTERVAL is pretty cheap (for queryable index, just a field access; for incremental index, getting the last key of a navigable map). So I'd be fine changing this to use the actual intervals.